File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/android/surfaceview16/ArgonGLView.java
package com.abubusoft.xenon.android.surfaceview16;

import com.abubusoft.xenon.opengl.XenonGLRenderer;

import android.content.Context;
import android.util.AttributeSet;
import android.view.SurfaceView;

public abstract class ArgonGLView extends SurfaceView {

	public ArgonGLView(Context context) {
		super(context);
	}
	
	public ArgonGLView(Context context, AttributeSet attrs) {
		super(context, attrs);
	}

	public abstract void onPause();
	
	public abstract void onResume();

	public abstract void setRenderer(XenonGLRenderer renderer);

	public abstract void setDebugFlags(int flags);

	public abstract void setEGLContextClientVersion(int version);

	public abstract void setPreserveEGLContextOnPause(boolean value);

	protected XenonGLRenderer mRenderer;
		
	public XenonGLRenderer getRenderer() {
		return mRenderer;
	}
		
}

File: xenon-android-library/src/main/kotlin/com/abubusoft/xenon/android/surfaceview16/ArgonGLView.kt
package com.abubusoft.xenon.android.surfaceview16

import android.content.Context
import android.util.AttributeSet
import android.view.SurfaceView
import com.abubusoft.xenon.opengl.XenonGLRenderer

abstract class ArgonGLView : SurfaceView {
    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {}

    abstract fun onPause()
    abstract fun onResume()
    abstract fun setRenderer(renderer: XenonGLRenderer)
    abstract fun setDebugFlags(flags: Int)
    abstract fun setEGLContextClientVersion(version: Int)
    abstract fun setPreserveEGLContextOnPause(value: Boolean)

    var renderer: XenonGLRenderer? = null
        protected set
}